src/Mvc/Mvc.ViewFeatures/src/RazorComponents/HtmlRenderer.cs

```diff
@@ -188,6 +194,34 @@ private int RenderFrames(HtmlRenderingContext context, ArrayRange<RenderTreeFram
         }
     }
 
+    private static int RenderTextareaElement(
+         HtmlRenderingContext context,
+         ArrayRange<RenderTreeFrame> frames,
+         int position)
+    {
+        ref var frame = ref frames.Array[position];
+        var result = context.HtmlContentBuilder;
+
+        result.AppendHtml("<");
+        result.AppendHtml(frame.ElementName);
+        _ = RenderAttributes(context, frames, position + 1, frame.ElementSubtreeLength - 1, false, out var capturedValueAttribute);
+        result.AppendHtml(">");
+
+        // Textarea is a special type of form field where the value is given as text content instead of a 'value' attribute
+        // So, if we captured a value attribute, use that instead of any child content
+        if (!string.IsNullOrEmpty(capturedValueAttribute))
+        {
+            result.Append(capturedValueAttribute);
+        }
+
+        var afterElement = position + frame.ElementSubtreeLength; // Skip descendants
+        result.AppendHtml("</");
+        result.AppendHtml(frame.ElementName);
+        result.AppendHtml(">");
+        Debug.Assert(afterElement == position + frame.ElementSubtreeLength);
+        return afterElement;
```

Comment:
Similar to above. Also we don't benefit from the `Debug.Assert` here because we know we just assigned that value, so there's nothing to check really:

```suggestion
        result.AppendHtml("</textarea>");
        return position + frame.ElementSubtreeLength; // Skip descendants
```